closed/src/java.base/share/native/libjncrypto/NativeCrypto.c

```diff
@@ -474,12 +576,272 @@ JNIEXPORT jlong JNICALL Java_jdk_crypto_jniprovider_NativeCrypto_loadCrypto
         }
     }
 
+    if (traceEnabled) {
+        fprintf(stdout, "%s: %s\n", message, openssl_version);
+        fflush(stdout);
+    }
+
+    return ossl_ver;
+}
+
+static void *
+load_crypto_library(jboolean traceEnabled, const char *libName)
+{
+    void *result = NULL;
+    if ((NULL != libName) && ('\0' != *libName)) {
+#if defined(_AIX)
+        int flags = RTLD_NOW;
+        if (NULL != strrchr(libName, '(')) {
+            flags |= RTLD_MEMBER;
+        }
+        result = dlopen(libName, flags);
+#elif defined(_WIN32) /* defined(_AIX) */
+        result = LoadLibrary(libName);
+#else /* defined(_WIN32) */
+        result = dlopen(libName, RTLD_NOW);
+#endif /* defined(_AIX) */
+    }
+    return result;
+}
+
+/* Look for a crypto library in java.home or the system.
+ * NULL is returned when an appropriate crypto library
+ * cannot be found.
+ */
+static void *
+find_crypto_library(jboolean traceEnabled, const char *chomepath)
+{
+    void *result = NULL;
+    void *prevResult = NULL;
+    size_t numOfLibs = 0;
+    size_t i = 0;
+    long tempVersion = 0;
+    long previousVersion = 0;
+    size_t num_of_generic = 0;
+
+#if defined(_AIX)
+    num_of_generic = 4;
+#elif defined(__linux__) /* defined(_AIX) */
+    num_of_generic = 1;
+#endif /* defined(_AIX) */
+
+    /* Library names for OpenSSL 1.1.1, 1.1.0 and symbolic links:
+     * It is important to preserve the order!!!
+     *
+     * Since there is no indication of the version of general
+     * symlink, they have to be loaded first, so as to compare
+     * with other available options.
+     *
+     * The rest of the library are in descending order, which allows
+     * us two things:
```

Comment:
Suggest the wording can be improved here (fixing typo: library -> libraries):
```
     * Since there is no indication of the version of a symlink,
     * they have to be loaded first, so as to compare with other
     * available options.
     *
     * The rest of the libraries are listed in descending order,
     * which allows us to do two things:
```

Reply:
Changed to suggestion.